Integrate swipe instruction

hsavoie opened the pull request:
Few things:

I put my SwipeListener class within the SwipeActivity class. Not sure if it would be better to have them separate?

I didn't bother with a button to reset it so it'll just count how many correct swipes you get until until you restart it, but I could easily put that in if we wanted.

Since there are only 4 directions, you may get the same one in a row multiple times, but all 4 should work if you test it long enough.

Again I'm not the best developer (I tried my best) so please suggest improvements!!! I'm sure there are ways to make this code more efficient and what not (ie enum)

hsavoie committed:
Integrate swipe instruction

Few things:

I put my SwipeListener class within the SwipeActivity class. Not sure if it would be better to have them separate?

I didn't bother with a button to reset it so it'll just count how many correct swipes you get until until you restart it, but I could easily put that in if we wanted.

Since there are only 4 directions, you may get the same one in a row multiple times, but all 4 should work if you test it long enough.

Again I'm not the best developer (I tried my best) so please suggest improvements!!! I'm sure there are ways to make this code more efficient and what not (ie enum)

hsavoie commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/SwipeActivity.java:
```diff
@@ -0,0 +1,141 @@
+        scoreText.setText("Score: " + score);
+    }
+
+    private class SwipeListener implements View.OnTouchListener{
```
Should I make this a separate class?

fredy20501 replied:
Yes, it will need to be a separate class since that is what we will use to integrate it with the main game loop.

hsavoie committed:
Update SwipeActivity.java

Commenting out text I added to test it

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/SwipeActivity.java:
```diff
@@ -0,0 +1,141 @@
+                    return true;
+                }
+            };
+
+            gestureDetector = new GestureDetector(listener);
```
This `new GestureDetector()` constructor is shown as being deprecated on my end. I think if we should use `GestureDetectorCompat()` instead, like this:
```java
gestureDetector = new GestureDetectorCompat(SwipeActivity.this, listener);

// Need to change the type of `gestureDetector` variable to GestureDetectorCompat
GestureDetectorCompat gestureDetector;
```

hsavoie replied:
Hmm, okay. I’ll look into this, thanks!

fredy20501 replied:
Actually, we can still use `GestureDetector()` but we need to use the version that takes in a context. So we should pass in a context into the SwipeListener constructor and use it here. Something like this:
```java
SwipeListener(Context context, View view){
    int SWIPE_THRESHOLD = 100;
    int SWIPE_VELOCITY_THRESHOLD = 100;

    // ...

    gestureDetector = new GestureDetector(context, listener);
    view.setOnTouchListener(this);
}
```
Then we pass in the context of the activity (`this`) when creating it:
```java
swipeListener = new SwipeListener(this, relativeLayout);
```
(Just like in the example you took the ideas from)

fredy20501 reviewed (commented):
Edit: Actually it seems fine the way it is. Don't mind me 😅 

~~The way you formatted it looks a little weird to me. I looked at the documentation for [Detecting Common Gestures](https://developer.android.com/training/gestures/detector#detect-a-subset-of-supported-gestures) and it looks a lot simpler than this. I'm guessing you might have followed an old tutorial that uses a different method.~~

~~(For example, I don't know why you implemented `View.OnTouchListener` instead of extending `GestureDetector.SimpleOnGestureListener` directly, like in the examples in the documentation)~~

carterjmoore commented:
Good work overall! Other than the changes Fred suggested, I guess I would just suggest maybe making an enum for the directions once you go to implement it with main project, as it will probably make things cleaner. That way, you can just have a private variable of the enum type, and set it when you randomly choose a direction, then compare directly to this value when a swipe is detected.

fredy20501 commented:
> Good work overall! Other than the changes Fred suggested, I guess I would just suggest maybe making an enum for the directions once you go to implement it with main project, as it will probably make things cleaner. That way, you can just have a private variable of the enum type, and set it when you randomly choose a direction, then compare directly to this value when a swipe is detected.

For the enum, you don't have to create your own. You can just use the `Instructions` enum that already exists and add your swipe directions to it (e.g., `SWIPE_RIGHT`, `SWIPE_LEFT`, ...).

hsavoie committed:
Improved based off feedback

- split classes: SwipeListener and SwipeActivity
- changed gestureDetector to take in context
- SwiperListener(this, relativeLayout)

Currently crashing. Not sure where I went wrong in splitting the classes

hsavoie committed:
Squashed commit of the following:

commit 90ee6345449ffd3cc02e0cf3580d382ba8384fae
Author: Frederic Verret <43451024+fredy20501@users.noreply.github.com>
Date:   Mon Feb 14 11:37:10 2022 -0400

    Implement Jump Instruction (#7)

commit 42f0f8d50d12c2473a567c0bd20ab798ed17f149
Author: Frederic Verret <43451024+fredy20501@users.noreply.github.com>
Date:   Mon Feb 14 11:31:58 2022 -0400

    Integrated shake instruction, organized java files (#5)

commit b6e6a6087fc61c63648557472a7eef86dcef4140
Author: Carter Moore <71897852+carterjmoore@users.noreply.github.com>
Date:   Sat Feb 12 11:47:11 2022 -0400

    Implemented game loop (#3)

    * Implemented game loop

    Set up the game loop. With every loop, we get a random instruction from a list of instructions and then add the corresponding UI elements. If the proper input is received, we clear the UI elements, wait one second, and then loop again. Also made it so the time available to perform the actions scales with score, but we might want to adjust this scale in the future.

    * Implemented changes Fred requested

    Added enums for Instructions, GameMode, and Difficulty. Instruction list is now constructed based on GameMode. Timer now pauses when app is paused and resumed when it resumes. Also cleaned up a few other things suggested by Fred.

    * Fixed a bug where "Play Again" caused crashing

    * Swapped MainActivity to be the main activity

commit d50e59e77b53a8895581894a2a81968915f21de9
Author: Frederic Verret <43451024+fredy20501@users.noreply.github.com>
Date:   Fri Feb 11 08:26:36 2022 -0400

    Implement ShakeDetector (#4)

    - Also created a ShakeTestActivity to test

hsavoie committed:
Separating classes (draft)

This is just a draft. I didnt finish what I was doing yet

fredy20501 committed:
Merge remote-tracking branch 'origin/main' into integrate-swipe

fredy20501 committed:
Finalize swipe instruction

fredy20501 committed:
Remove old changes from before the refactor

carterjmoore reviewed (approved):
Looks nice 💯 🥇 🎆

fredy20501 committed:
Merged Tap/Swipe into TouchDetector

carterjmoore commented:
Conflicts 😩 😖 😕

fredy20501 committed:
Merge remote-tracking branch 'origin/main' into integrate-swipe

fredy20501 committed:
Fix freeze & rotate instructions
